doc/user/spatial_block_parameters.rst

```diff
@@ -0,0 +1,339 @@
+3. The number of pins is determined by :meth:`armi.reactor.blocks.HexBlock.getNumPins` which may place similar
+   restrictions on what constitutes a pin.
+
+If the auto grid creation is successful, components with a multiplicity equal to the number of pins will be assigned
+locations on the lattice grid.
+
+.. warning::
+
+    Consider subclassing :class:`~armi.reactor.blocks.HexBlock` with specific pin-like methods and
+    overriding the :meth:`~armi.reactor.blocks.HexBlock.autoCreateSpatialGrids` if you want complete control
+    over this process. Alternatively, use an explicit grid in blueprints.
+
+
+Interacting with spatial data
+=============================
+
+This section will focus on accessing locations of components in the block, locations of specifically
+pins, and examples of some pin data that may be assigned to a block's parameter set.
+
+Component locations
+-------------------
+
+Components that live on a spatial grid have a ``spatialLocator`` attribute to help indicate where that
+component exists in space. If we grab the fuel component from the UO2 block in the
+:ref:`ANL AFCI 177 example <walkthrough-inputs>` we can see where it exists in the block::
+
+    >>> import armi
+    >>> armi.configure()
+    >>> from armi.reactor.flags import Flags
+    >>> r = armi.init(fName="c5g7-settings.yaml").r
+    >>> fuelAssem = r.core[5]
+    >>> fuelBlock = fuelAssem[1]
+    >>> fuelBlock.spatialGrid
+    <HexGrid -- 2046645914880
+    Bounds:
+    None
+    None
+    None
+    Steps:
+    [ 0.4444 -0.4444  0.    ]
+    [0.76972338 0.76972338 0.        ]
+    [0. 0. 0.]
+    Anchor: <fuel B0009-001 at 008-040-001 XS: C BU GP: A>
+    Offset: [0. 0. 0.]
+    Num Locations: 400>
+    >>> fuel = fuelBlock.getChildrenWithFlags(Flags.FUEL)[0]
+    >>> fuel.getDimension("mult")
+    271
+    >>> fuel.spatialLocator
+    <MultiIndexLocation with 271 locations>
+
+This :class:`~armi.reactor.grids.MultiIndexLocation` is a way to indicate this Component exists at multiple
+sites. Each item in this locator is one location on the underlying grid where we could find this component::
+
+    >>> fuel.spatialLocator[0]
+    <IndexLocation @ (0,0,0)>
+    >>> fuel.spatialLocator[0].getLocalCoordinates()
+    array([0., 0., 0.])
+    >>> coordsFromFuel = fuel.spatialLocator.getLocalCoordinates()
+    >>> coordsFromFuel.shape
+    (271, 3)
+    >>> coordsFromFuel
+    array([[ 0.        ,  0.        ,  0.        ],
+           [ 0.4444    ,  0.76972338,  0.        ],
+           [-0.4444    ,  0.76972338,  0.        ],
+           [-0.8888    ,  0.        ,  0.        ],
+    ...
+           [ 5.7772    ,  3.84861689,  0.        ],
+           [ 5.3328    ,  4.61834027,  0.        ],
+           [ 4.8884    ,  5.38806365,  0.        ],
+           [ 4.444     ,  6.15778703,  0.        ]])
+
+We get a ``(271, 3)`` array because we have 271 of these fuel components in the block, and each row contains one
```

Comment:
I think this would be cleaner and easier to read if you just did `coordsFromFuel.shape` or something. We don't think the users are going to be spot checking these values here. What we want is to show them the dimensionality of this example.

Just my two cents.